pyadjoint/overloaded_type.py

```diff
@@ -142,6 +142,17 @@ def _ad_restore_at_checkpoint(self, checkpoint):
         """
         raise NotImplementedError
 
+    def _ad_assign(self, other):
+        """This method must be overridden.
+
+        The method should implement a routine for assigning the values from
+        another object `other` to `self`.
+
+        Args:
+            value (obj): The object which should be assigned to `self`.
+        """
+        raise NotImplementedError
+
```

Comment:
I am stealing your idea.

Reply:
Only for needed for mutable types (`AdjFloat is immutable).